src/poetry/puzzle/provider.py

```diff
@@ -230,7 +236,13 @@ def search_for_vcs(self, dependency: VCSDependency) -> list[Package]:
         Basically, we clone the repository in a temporary directory
         and get the information we need by checking out the specified reference.
         """
-        if dependency in self._deferred_cache:
+        # we ensure subdirectory match here as workaround until poetry-core is updated
+        # to >1.1.0a7
+        if (
+            dependency in self._deferred_cache
+            and self._deferred_cache[dependency].source_subdirectory
```

Comment:
This is another case that will be removed once core is upgraded.

Reply:
The same as comment above, please add a `#FIXME` comment.